qiskit/transpiler/passes/layout/vf2_utils.py

```diff
@@ -98,8 +111,9 @@ def score_layout(
         size = 0
     nlayout = NLayout(layout_mapping, size + 1, size + 1)
     bit_list = np.zeros(len(im_graph), dtype=np.int32)
-    for node_index in bit_map.values():
-        bit_list[node_index] = sum(im_graph[node_index].values())
+    if strict_direction:
+        for node_index in bit_map.values():
+            bit_list[node_index] = sum(im_graph[node_index].values())
```

Comment:
This (and the bit in `build_average_error_map`) look like a slightly separate bugfix.  Is that right?

Reply:
Yeah, although it's semi related I think (it's been a while since I looked at this patch). The strict direction case doesn't use the common base between the passes in the same way (since vf2post has to do a lot more checking to ensure direcitonality constraints and target basis are preserved). Splitting out the 1q piece from things broke things IIRC (only vaguely remember) so we needed the if condition to branch the logic on that otherwise it failed.